table/tables/tables.go

```diff
@@ -1480,20 +1481,39 @@ func CheckHandleOrUniqueKeyExistForUpdateIgnoreOrInsertOnDupIgnore(ctx context.C
 
 	// Check unique key exists.
 	{
-		shouldSkipIgnoreCheck := func(idx table.Index) bool {
+		shouldSkipIgnoreCheck := func(idx table.Index) (bool, error) {
 			if !IsIndexWritable(idx) || !idx.Meta().Unique || (t.Meta().IsCommonHandle && idx.Meta().Primary) {
-				return true
+				return true, nil
 			}
 			for _, c := range idx.Meta().Columns {
 				if modified[c.Offset] {
-					return false
+					if c.Length != types.UnspecifiedLength && (newRow[c.Offset].Kind() == types.KindString || newRow[c.Offset].Kind() == types.KindBytes) {
```

Comment:
Besides `KindString` the Will other prefix index column types need similar process like this?

Reply:
Agree, feel that there may still be problems later, but this type of SQL execution is rare, so it may okay for this release branch.